RulesEngine/RuleProcessor.cs

```diff
@@ -17,242 +16,235 @@ namespace Microsoft.ApplicationInspector.RulesEngine
+                                    { 
+                                        //restrict tags build files to tags with "metadata" to avoid false feature positives that are not part of executable code
+                                        if (languageInfo.Type == LanguageInfo.LangFileType.Build && orh.AppInspectorRule.Tags.Any(v => !v.Contains("Metadata")))
                                         {
-                                            passedConditions = BestMatch(resultsList, newMatch);//check all rules in permanent list
+                                            continue;
                                         }
-                                    }
-                                }
-                            }
 
-                            if (passedConditions)
-                            {
-                                matchList.Add(newMatch);
+                                        if (ConfidenceLevelFilter.HasFlag(pattern.Confidence))
+                                        {
+                                            ScanResult newMatch = new ScanResult()
+                                            {
+                                                Boundary = boundary,
+                                                StartLocation = textContainer.GetLocation(boundary.Index),
+                                                EndLocation = textContainer.GetLocation(boundary.Index + boundary.Length),
+                                                PatternMatch = pattern,
+                                                Confidence = pattern.Confidence,
+                                                Rule = orh.AppInspectorRule
+                                            };
+
+                                            if (_uniqueTagMatchesOnly)
+                                            {
+                                                if (RuleTagsAreUniqueOrAllowed(newMatch.Rule.Tags))
+                                                {
+                                                    if (!_stopAfterFirstMatch) //use first or best match options
+                                                    {
+                                                        if (BestMatch(newMatches, newMatch) && BestMatch(resultsList, newMatch))
+                                                        {
+                                                            newMatches.Add(newMatch); //replaces previous and now removed match
+                                                        }
+                                                    }
+                                                    else
+                                                    {
+                                                        newMatches.Add(newMatch); //first unique match and we're done
```

Comment:
This doesn't look right to me.  This adds the full set of new matches the same as the below else statement does.